File: sgf-poker/sgf-poker-ios/Core/Repositories/GameRepository.swift
// Core/Repositories/GameRepository.swift

import Foundation

/// Contract for any games data source.
/// Conforming types can be in-memory, CoreData, network, etc.
protocol GameRepository: AnyObject {
    /// Returns all stored games, sorted by creation date descending.
    func fetchAll() throws -> [Game]

    /// Persists a new game. Throws if a game with the same name exists.
    func save(_ game: Game) throws

    /// Removes a game by its identifier.
    func delete(id: UUID) throws
}

File: sgf-poker-android/app/src/main/java/com/sgf/poker/data/repository/GameRepository.java
package com.sgf.poker.data.repository;

import com.sgf.poker.domain.model.Game;
import java.util.List;

/** Contract for any games data source. */
public interface GameRepository {
    /** Returns all games sorted by eventDate descending (most recent first). */
    List<Game> fetchAll();

    /** Persists a game. Throws GameException.duplicateDate if month/year already exists. */
    void save(Game game);

    /** Removes a game by id. Throws GameException.notFound if missing. */
    void delete(String id);
}
